### DISCUSSION TITLE
Postgres Package Manager

### DISCUSSION THREAD

**Discussion by psteinroe** [supabase/supabase#9219] [Feature Requests] [answered]
Hey,

supabase does a great job at promoting pg as your “application server” where most of the business logic runs within functions and triggers. Working with it for a while now, I developed a few functions that I think might be useful for others, e.g. handlebar template parsing or sending request to the segment API. As of now, I could create a repo or gist for others to copy-and-paste the code into their migrations. Of course, there is a better solution: A npm-like package manager for postgres.

There are few major challenges that come with it and I am not very experienced with writing postgres extensions, but would love to exchange ideas. Currently, this is my basic idea on how to do this:

- A package would define its scope (either manually or automatically), e.g. functions with a specific signature or entire schemas
- The access to these “managed” objects is restricted
- All package upgrades are shipped as migrations, and the scope of a package can change with each one

I heard that a pg package manager has been the initial idea for the supabase company, and I would love to bring that idea back!

**Comment by thorwebdev** [comment]
Funnily enough this was one of the original ideas behind Supabase and someone we're still exploring. We'll rope you in when we get a bit further along on this 👍

**Comment by inian** [comment, marked as answer]
We have launched https://database.dev/ for postgres package management
